### PR TITLE
Extract Files task fails: not enough memory

### PR DESCRIPTION
**Pull request by EzzhevNikita**
**Task name**: ExtractFiles 

**Description**: Added support for content overwriting in output directory, using additional input parameter
[#120](https://github.com/microsoft/build-task-team/issues/120) 
[Feedback ticket](https://dev.azure.com/mseng/AzureDevOps/_workitems/edit/1686882)

I applied some amendments to existing pr's that related to this issue and several related ones and created this one

**Documentation changes required:** (Y/N) Y

**Added unit tests:** (Y/N) Y

**Attached related issue:** (Y/N) Y
[Related Issue 1](https://github.com/microsoft/azure-pipelines-tasks/issues/11082)
[Related Issue 2](https://github.com/microsoft/azure-pipelines-tasks/issues/11593)

**Related ps's:**
[Related PR 1](https://github.com/microsoft/azure-pipelines-tasks/pull/6760)
[Related PR 2](https://github.com/microsoft/azure-pipelines-tasks/pull/11594)

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

File: Tasks/ExtractFilesV1/extractfilestask.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ import os = require('os');
 var archiveFilePatterns: string[] = tl.getDelimitedInput('archiveFilePatterns', '\n', true);
 var destinationFolder: string = path.normalize(tl.getPathInput('destinationFolder', true, false).trim());
 var cleanDestinationFolder: boolean = tl.getBoolInput('cleanDestinationFolder', false);
+var overwriteExistingFiles: boolean = tl.getBoolInput('overwriteExistingFiles', false);
```
Comment (anton-erofeev) [inline Tasks/ExtractFilesV1/extractfilestask.ts:11]: Can we use `let` or `const` here?

Comment (EzzhevNikita) [reply Tasks/ExtractFilesV1/extractfilestask.ts:11]: Yeah, we can use `const` here, `var` were used just for consistency with other variables declaration for input fields. Should I change it just for new variable only or for all of them?

**Review by anatolybolshakov — APPROVED** [review]
LGTM, thanks!

File: Tasks/ExtractFilesV1/extractfilestask.ts (hunk 1 above)
Comment (anatolybolshakov) [reply Tasks/ExtractFilesV1/extractfilestask.ts:11]: Thanks @anton-erofeev for the point - let's just leave it as is for consistency here

**Review by anton-erofeev — APPROVED** [review]

**Review by damccorm — APPROVED** [review]
LGTM.

2 notes though:
1) Please make sure CI is passing before merging. Its currently broken due to an agent issue, but it should be back up by the end of the day.
2) Could you name the commit something a little more descriptive of the change itself? (e.g. "Extract Files task: add overwrite option")

**Pull request merged**